operator/internal/handlers/lokistack_check_cert_expiry.go

```diff
@@ -0,0 +1,55 @@
+package handlers
+
+import (
+	"context"
+
+	"github.com/ViaQ/logerr/v2/kverrors"
+	"github.com/go-logr/logr"
+	configv1 "github.com/grafana/loki/operator/apis/config/v1"
+	lokiv1 "github.com/grafana/loki/operator/apis/loki/v1"
+	"github.com/grafana/loki/operator/internal/certrotation"
+	"github.com/grafana/loki/operator/internal/external/k8s"
+	"github.com/grafana/loki/operator/internal/handlers/internal/certificates"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// CheckCertExpiry handles the case if the LokiStack managed signing CA, client and/or serving
+// certificates expired. Returns true if any of those expired and an error representing the reason
+// of expiry.
+func CheckCertExpiry(ctx context.Context, log logr.Logger, req ctrl.Request, k k8s.Client, fg configv1.FeatureGates) error {
+	ll := log.WithValues("lokistack", req.String(), "event", "checkCertExpiry")
+
+	var stack lokiv1.LokiStack
+	if err := k.Get(ctx, req.NamespacedName, &stack); err != nil {
```

Comment:
Why not declare the variable as a pointer already?

```suggestion
	var stack *lokiv1.LokiStack
	if err := k.Get(ctx, req.NamespacedName, stack); err != nil {
```

Reply:
There is a limitation or error that I have not digged into it since the inception of this code base. Switching to a pointer here fails with using reflection in the tests where we use counterfeiter:
```
panic: reflect: call of reflect.Value.Set on zero Value [recovered]
	panic: reflect: call of reflect.Value.Set on zero Value
```
This is something for another day.